core/src/main/scala/cats/Apply.scala

```diff
@@ -19,6 +19,14 @@ trait Apply[F[_]] extends Functor[F] with Cartesian[F] with ApplyArityFunctions[
   override def product[A, B](fa: F[A], fb: F[B]): F[(A, B)] =
     ap(map(fa)(a => (b: B) => (a, b)))(fb)
 
+  /** Sequentially compose two actions, discarding any value produced by the first. */
+  def followedBy[A, B](fa: F[A])(fb: F[B]): F[B] =
+    map(product(fa, fb)) { case (_, b) => b }
```

Comment:
Shouldn’t we use map2 here since that can sometimes be more efficient than product + map

Reply:
Yes that's a great idea

Reply:
also I think "sequentially" is confusing now. It is sequential if we have a monad, if we have an applicative it is not necessarily "sequential"